app/main/views/suppliers.py

```diff
@@ -119,6 +119,38 @@ def view_signed_agreement(supplier_id, framework_slug):
     )
 
 
+@main.route('/suppliers/agreements/<agreement_id>/on-hold', methods=['POST'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def put_signed_agreement_on_hold(agreement_id):
+    data_api_client.put_signed_agreement_on_hold(agreement_id, current_user.email_address)
+
+    organisation = request.args['organisation']
+    flash('The agreement for {} was put on hold.'.format(organisation), 'message')
```

Comment:
Yeah, I've put them in in anticipation of the route redirecting to another agreement page instead of the list view. It's only pointing at the list view atm cos the filtering isn't in place to get the next agreement.